core/src/main/java/feast/core/grpc/CoreServiceImpl.java

```diff
@@ -107,6 +107,34 @@ public void listFeatureSets(
     }
   }
 
+  @Override
+  public void listFeatures(
+      ListFeaturesRequest request, StreamObserver<ListFeaturesResponse> responseObserver) {
+    try {
+      ListFeaturesResponse response = specService.listFeatures(request.getFilter());
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (RetrievalException | IllegalArgumentException | InvalidProtocolBufferException e) {
+      log.error("Exception has occurred in ListFeatures method: ", e);
+      responseObserver.onError(
+          Status.INTERNAL.withDescription(e.getMessage()).withCause(e).asRuntimeException());
+    }
+  }
+
+  @Override
+  public void listEntities(
+      ListEntitiesRequest request, StreamObserver<ListEntitiesResponse> responseObserver) {
+    try {
+      ListEntitiesResponse response = specService.listEntities(request.getFilter());
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (RetrievalException | IllegalArgumentException | InvalidProtocolBufferException e) {
```

Comment:
Is it possible for you to return different gRPC exceptions here instead of INTERNAL for everything? Also, we should probably also catch and return unknown exceptions for the time being, because it can be hard to debug otherwise.